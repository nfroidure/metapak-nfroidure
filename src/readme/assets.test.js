'use strict';

const assert = require('assert');
const sinon = require('sinon');
const assetsTransformer = require('./assets');

describe('Assets transformer for www configs', () => {
  it('should build the README.md file', (done) => {
    const fs = {
      readFileAsync: sinon.stub(),
    };
    const PROJECT_DIR = '/lol/';
    const log = {
      error: sinon.stub,
    };

    fs.readFileAsync.onFirstCall().returns(Promise.resolve('## Usage\nJust require me\n'));
    fs.readFileAsync.onSecondCall().returns(Promise.resolve(''));

    assetsTransformer({
      name: 'README.md',
      data: '<!-- something -->\n',
    }, {
      name: 'module',
      description: 'A great module!',
      devDependencies: {},
      license: 'MIT',
    }, {
      PROJECT_DIR,
      fs,
      log,
    })
    .then((file) => {
      assert.deepEqual(
        file,
        {
          name: 'README.md',
          data: '<!-- something -->\n' +
          '# module\n' +
          '> A great module!\n\n' +
          '[![NPM version](https://badge.fury.io/js/module.svg)]' +
          '(https://npmjs.org/package/module)\n' +
          '[![Build status](https://secure.travis-ci.org/nfroidure/module.svg)]' +
          '(https://travis-ci.org/nfroidure/module)\n' +
          '[![Dependency Status](https://david-dm.org/nfroidure/module.svg)]' +
          '(https://david-dm.org/nfroidure/module)\n' +
          '[![devDependency Status](https://david-dm.org/nfroidure/module/dev-status.svg)]' +
          '(https://david-dm.org/nfroidure/module#info=devDependencies)\n' +
          '[![Code Climate](https://codeclimate.com/github/nfroidure/module.svg)]' +
          '(https://codeclimate.com/github/nfroidure/module)\n' +
          '[![Dependency Status](https://dependencyci.com/github/nfroidure/module/badge)]' +
          '(https://dependencyci.com/github/nfroidure/module)\n\n' +
          '## Usage\n' +
          'Just require me\n\n' +
          '# License\n' +
          '[MIT](https://github.com/nfroidure/module/blob/master/LICENSE)\n',
        }
      );
    })
    .then(done)
    .catch(done);
  });

  it('should let pass other files', () => {
    assert.deepEqual(
      assetsTransformer({
        name: 'YOLO',
        data: 'Carpe diem\n',
      }, {}, {}),
      {
        name: 'YOLO',
        data: 'Carpe diem\n',
      }
    );
  });
});
