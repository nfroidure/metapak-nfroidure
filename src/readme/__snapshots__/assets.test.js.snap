// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assets transformer for www configs should build the README.md file 1`] = `
Object {
  "data": "[//]: # ( )
[//]: # (This file is automatically generated by a \`metapak\`)
[//]: # (module. Do not change it except between the)
[//]: # (\`content:start/end\` flags, your changes would)
[//]: # (be overridden.)
[//]: # ( )
# module
> A great module!

[![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/nfroidure/module/blob/master/LICENSE)
[![NPM version](https://badge.fury.io/js/module.svg)](https://npmjs.org/package/module)
[![Dependency Status](https://david-dm.org/nfroidure/module.svg)](https://david-dm.org/nfroidure/module)
[![devDependency Status](https://david-dm.org/nfroidure/module/dev-status.svg)](https://david-dm.org/nfroidure/module#info=devDependencies)
[![Package Quality](http://npm.packagequality.com/shield/module.svg)](http://packagequality.com/#?package=module)


[//]: # (::contents:start)

## Usage

Just require me


[//]: # (::contents:end)


# API
\`\`\`
toto.lol()
\`\`\`

# Authors
- [Nicolas Froidure](http://insertafter.com/en/index.html)
- John Doe

# License
[MIT](https://github.com/nfroidure/module/blob/master/LICENSE)
",
  "name": "README.md",
}
`;

exports[`Assets transformer for www configs should build the README.md file with additional badges 1`] = `
Object {
  "data": "[//]: # ( )
[//]: # (This file is automatically generated by a \`metapak\`)
[//]: # (module. Do not change it except between the)
[//]: # (\`content:start/end\` flags, your changes would)
[//]: # (be overridden.)
[//]: # ( )
# module
> A great module!

[![GitHub license](https://img.shields.io/badge/license-MIT-blue.svg)](https://github.com/nfroidure/module/blob/master/LICENSE)
[![Build status](https://secure.travis-ci.org/nfroidure/module.svg)](https://travis-ci.org/nfroidure/module)
[![Coverage Status](https://coveralls.io/repos/nfroidure/module/badge.svg?branch=master)](https://coveralls.io/r/nfroidure/module?branch=master)
[![NPM version](https://badge.fury.io/js/module.svg)](https://npmjs.org/package/module)
[![Dependency Status](https://david-dm.org/nfroidure/module.svg)](https://david-dm.org/nfroidure/module)
[![devDependency Status](https://david-dm.org/nfroidure/module/dev-status.svg)](https://david-dm.org/nfroidure/module#info=devDependencies)
[![Package Quality](http://npm.packagequality.com/shield/module.svg)](http://packagequality.com/#?package=module)
[![Code Climate](https://codeclimate.com/github/nfroidure/module.svg)](https://codeclimate.com/github/nfroidure/module)


[//]: # (::contents:start)

## Usage

Just require me


[//]: # (::contents:end)


# API
\`\`\`
toto.lol()
\`\`\`

# Authors
- [Nicolas Froidure](http://insertafter.com/en/index.html)
- John Doe

# License
[MIT](https://github.com/nfroidure/module/blob/master/LICENSE)
",
  "name": "README.md",
}
`;

exports[`Assets transformer for www configs should build the README.md file with no badge 1`] = `
Object {
  "data": "[//]: # ( )
[//]: # (This file is automatically generated by a \`metapak\`)
[//]: # (module. Do not change it except between the)
[//]: # (\`content:start/end\` flags, your changes would)
[//]: # (be overridden.)
[//]: # ( )
# module
> A great module!



[//]: # (::contents:start)

## Usage

Just require me


[//]: # (::contents:end)


# API
\`\`\`
toto.lol()
\`\`\`

# Authors
- [Nicolas Froidure](http://insertafter.com/en/index.html)
- John Doe

# License
[MIT](https://github.com/nfroidure/module/blob/master/LICENSE)
",
  "name": "README.md",
}
`;
